src/components/AcceptDialog.tsx

```diff
@@ -12,13 +12,38 @@ import {
 import { DocumentState } from '../domain/document-submission';
 import DateFields from './DateFields';
 import router from 'next/router';
+import { DateTime } from 'luxon';
 
 const schema = Yup.object().shape({
   staffSelectedDocumentTypeId: Yup.string().required(
     'Please choose a document type'
   ),
+  validUntilDates: Yup.array().test(
+    'is-date',
+    'Please enter a valid date',
+    (values) => isDate(values)
+  ),
 });
 
+const isDate = (
+  validUntilDates: string[] | null | undefined | unknown[]
+): boolean => {
+  if (
+    validUntilDates?.every(
+      (x: string | null | undefined | unknown) => x == undefined
+    )
+  )
+    return true;
+
+  const concatDates = validUntilDates?.join('-');
+  if (concatDates) {
+    const parseDate = DateTime.fromFormat(concatDates, 'd-L-yyyy');
+    return !parseDate?.invalidReason;
```

Comment:
If the date cannot be created, then an invalid reason will be created -- this will return false and make the 'test' fail and throw the error.